opentracing-spring-web/src/test/java/io/opentracing/contrib/spring/web/client/TracingAsyncRestTemplateTest.java

```diff
@@ -0,0 +1,43 @@
+package io.opentracing.contrib.spring.web.client;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.AsyncClientHttpRequestInterceptor;
+import org.springframework.test.web.client.MockRestServiceServer;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.web.client.AsyncRestTemplate;
+
+import java.util.Collections;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * @author Pavol Loffay
+ */
+public class TracingAsyncRestTemplateTest extends AbstractTracingClientTest<AsyncRestTemplate> {
```

Comment:
You need far more tests for the async template than for the regular sync one :) Please, add more tests that exercise the async features.